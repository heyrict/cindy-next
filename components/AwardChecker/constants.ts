export const PuzzleCountAwards = {
  190: 1,
  191: 5,
  192: 10,
  193: 30,
  194: 50,
  14: 100,
  15: 200,
  16: 300,
  17: 500,
  18: 600,
  19: 800,
  20: 1000,
  21: 1500,
  22: 2000,
};

export const QuestionCountAwards = {
  195: 10,
  196: 50,
  197: 100,
  198: 300,
  199: 500,
  24: 1000,
  25: 2000,
  26: 3000,
  27: 5000,
  28: 6000,
  29: 8000,
  30: 10000,
  31: 15000,
  32: 20000,
  33: 30000,
};

export const TrueAnswerCountAwards = {
  175: 1,
  176: 5,
  177: 11,
  34: 33,
  35: 55,
  36: 77,
  178: 111,
  37: 222,
  38: 333,
  39: 555,
  40: 777,
  41: 1111,
  42: 2222,
  179: 3333,
};

export const GoodQuestionCountAwards = {
  171: 11,
  43: 55,
  44: 111,
  45: 333,
  172: 555,
  173: 777,
  46: 1111,
  47: 2222,
  48: 3333,
  49: 5555,
  50: 7777,
  51: 11111,
  52: 22222,
  174: 33333,
};

export const StarSumAwards = {
  181: 10,
  182: 30,
  183: 50,
  184: 100,
  185: 300,
  186: 500,
  187: 1000,
  188: 3000,
  189: 10000,
};

export const StarByPuzzleAwards = {
  101: { starCount: 10, puzzleCount: 1 },
  102: { starCount: 10, puzzleCount: 2 },
  103: { starCount: 20, puzzleCount: 1 },
  104: { starCount: 20, puzzleCount: 2 },
  105: { starCount: 30, puzzleCount: 1 },
  106: { starCount: 30, puzzleCount: 2 },
  107: { starCount: 50, puzzleCount: 1 },
  108: { starCount: 50, puzzleCount: 2 },
  109: { starCount: 75, puzzleCount: 1 },
  110: { starCount: 75, puzzleCount: 2 },
  111: { starCount: 100, puzzleCount: 1 },
  112: { starCount: 100, puzzleCount: 2 },
  113: { starCount: 100, puzzleCount: 3 },
  114: { starCount: 100, puzzleCount: 4 },
};

export const PuzzleByGenreAwards = {
  200: { genre: 0, count: 10 },
  201: { genre: 0, count: 30 },
  202: { genre: 0, count: 50 },
  203: { genre: 1, count: 5 },
  204: { genre: 1, count: 10 },
  205: { genre: 1, count: 15 },
  206: { genre: 2, count: 1 },
  207: { genre: 2, count: 5 },
  208: { genre: 2, count: 10 },
  209: { genre: 3, count: 1 },
  210: { genre: 3, count: 5 },
  211: { genre: 3, count: 10 },
};

export const PuzzleByYamiAwards = {
  212: 5,
  213: 10,
  214: 15,
};

export const PuzzleByYamiQuestionsAwards = {
  217: 300,
  218: 500,
  219: 1000,
};

export const MixedAwards = {
  180: { questionCount: 10000, puzzleCount: 100 },
};

export const SpecialAwards = [115, 215, 216];
