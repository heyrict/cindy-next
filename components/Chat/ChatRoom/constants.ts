export const CHATMESSAGES_PER_PAGE = 10;
