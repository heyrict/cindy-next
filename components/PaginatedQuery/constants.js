export const PB_ORIENT = {
  none: 0,
  bottom: 1,
  top: 2,
  both: 3,
};
