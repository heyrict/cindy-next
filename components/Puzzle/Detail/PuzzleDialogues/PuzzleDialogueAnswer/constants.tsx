export const AnswerModes = {
  DISPLAY: Symbol('display'),
  EDIT: Symbol('edit'),
};
