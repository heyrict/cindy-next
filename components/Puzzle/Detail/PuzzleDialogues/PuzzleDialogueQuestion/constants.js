export const QuestionModes = {
  DISPLAY: Symbol('display'),
  EDIT: Symbol('edit'),
};
