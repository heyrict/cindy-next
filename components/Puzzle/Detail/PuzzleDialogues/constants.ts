import { ExtractUserFilterUserReturnType } from './types';
import {
  DialogueHintQuery_puzzleLogs_Hint,
  DialogueHintQuery_puzzleLogs_Dialogue,
} from 'graphql/Queries/generated/DialogueHintQuery';

type Dialogue = DialogueHintQuery_puzzleLogs_Dialogue;
type Hint = DialogueHintQuery_puzzleLogs_Hint;

export const extractUserFilterUserFromDialogues = (
  dialogues: Array<Dialogue | Hint>,
): Array<ExtractUserFilterUserReturnType> => {
  const users: {
    [userId: number]: ExtractUserFilterUserReturnType;
  } = {};
  dialogues.forEach(dialogue => {
    if (dialogue.__typename == 'Hint') return;

    if (!(dialogue.user.id in users)) {
      users[dialogue.user.id] = {
        id: dialogue.user.id,
        nickname: dialogue.user.nickname,
        dialogueCount: 1,
        answeredDialogueCount: dialogue.answer === '' ? 0 : 1,
        dialogueHasTrue: dialogue.true,
      };
      return;
    }

    users[dialogue.user.id].dialogueCount += 1;
    if (dialogue.answer !== '')
      users[dialogue.user.id].answeredDialogueCount += 1;
    if (dialogue.true) users[dialogue.user.id].dialogueHasTrue = true;
  });
  return Object.values(users);
};
