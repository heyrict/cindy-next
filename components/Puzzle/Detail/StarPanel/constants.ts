export const MAX_DISPLAY_STAR_USERS = 12;
