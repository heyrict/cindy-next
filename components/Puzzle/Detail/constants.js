export const widthSplits = [
  [1, '3%', '5%', '8%'],
  [2, '5%', '8%', '12%'],
  [3, '6%', '10%', '15%'],
];
