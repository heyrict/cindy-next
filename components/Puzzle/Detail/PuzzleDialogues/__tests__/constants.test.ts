import { extractUserFilterUserFromDialogues } from '../constants';

import { dialogues } from './constants';

describe('function extractUserFilterUserFromDialogues()', () => {
  it('Dialogues should be filtered properly', () => {
    const result = extractUserFilterUserFromDialogues(dialogues);
    expect(result).toMatchSnapshot();
  });
});
