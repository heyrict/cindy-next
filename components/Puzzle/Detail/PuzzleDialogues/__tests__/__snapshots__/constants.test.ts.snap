// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function extractUserFilterUserFromDialogues() Dialogues should be filtered properly 1`] = `
Array [
  Object {
    "dialogueCount": 1,
    "dialogueHasTrue": false,
    "dialogueUnsolvedCount": 0,
    "id": 13,
    "nickname": "Bar",
  },
  Object {
    "dialogueCount": 2,
    "dialogueHasTrue": false,
    "dialogueUnsolvedCount": 0,
    "id": 73,
    "nickname": "Foo",
  },
]
`;
